### ISSUE TITLE
loss_box_reg 

### ISSUE THREAD

**Issue by pierrelzw** [asharakeh/pod_compare#4]
https://github.com/asharakeh/pod_compare/blob/7784b0b45fc6e6ee17e87132c75fa9583004e57f/src/probabilistic_modeling/probabilistic_retinanet.py#L298

Hi  @asharakeh, thanks for sharing your work! According to your paper, the loss with bbox_cov prediction should be 

![image](https://user-images.githubusercontent.com/13348529/104911057-3bb3ca80-59c5-11eb-8b94-4e60f06e47cc.png)

but the implementation below seems to be :
![image](https://user-images.githubusercontent.com/13348529/104911188-6ef65980-59c5-11eb-8e35-859002e26a2d.png)

It seems they are not equal. Am I miss something?

```  
                    loss_box_reg = 0.5 * torch.exp(-pred_bbox_cov) * smooth_l1_loss(
                        pred_anchor_deltas,
                        gt_anchors_deltas,
                        beta=self.smooth_l1_beta)

                    loss_covariance_regularize = 0.5 * pred_bbox_cov
                    loss_box_reg += loss_covariance_regularize

                    loss_box_reg = torch.sum(
                        loss_box_reg) / max(1, self.loss_normalizer)
```

**Comment by asharakeh** [comment]
Hi, the implementation I provide is correct.

Here are the shapes of the torch.exp(-pred_bbox_cov) and smooth_l1_loss()....

![1](https://user-images.githubusercontent.com/19838254/104948335-228e2800-598b-11eb-916b-4c4d368a1b7b.png)
![2](https://user-images.githubusercontent.com/19838254/104948347-24f08200-598b-11eb-9ae6-554484dc2fce.png)

**Comment by asharakeh** [comment]
Basically torch.exp(-pred_bbox_cov) and smooth_l1_loss() both have an Nx4 shape. You multiply them first, then apply the sum later in: 

`loss_box_reg = torch.sum(
                        loss_box_reg) / max(1, self.loss_normalizer)`

**Comment by pierrelzw** [comment]
Hi you are right.  The reduction is not applied in smooth_l1_loss. Thanks！

**Comment by pierrelzw** [comment]
> Hi, the implementation I provide is correct.
> ![1](https://user-images.githubusercontent.com/19838254/104948437-4baeb880-598b-11eb-8aec-8f0228660e82.png)
> ![2](https://user-images.githubusercontent.com/19838254/104948442-4cdfe580-598b-11eb-9586-a6801c95e899.png)

What's the tool you use here to estimate the shape of Torch Tensor? It's cool !

**Comment by asharakeh** [comment]
It is the pycharm IDE https://www.jetbrains.com/pycharm/.
